packages/radio-player/src/radio-player.ts

```diff
@@ -0,0 +1,738 @@
+/* eslint-disable import/no-duplicates */
+import {
+  LitElement,
+  html,
+  css,
+  customElement,
+  property,
+  CSSResult,
+  TemplateResult,
+  PropertyValues,
+} from 'lit-element';
+import { AudioElement, AudioSource } from '@internetarchive/audio-element';
+import {
+  TranscriptConfig,
+  TranscriptEntryConfig,
+  TranscriptView,
+} from '@internetarchive/transcript-view';
+
+import '@internetarchive/waveform-progress';
+import '@internetarchive/playback-controls';
+import '@internetarchive/scrubber-bar';
+
+import { QuickSearchEntry } from '@internetarchive/expandable-search-bar';
+import '@internetarchive/expandable-search-bar';
+
+import './search-results-switcher';
+
+import { ZoneOfSilence } from '@internetarchive/waveform-progress';
+import { PlaybackControls, PlaybackMode } from '@internetarchive/playback-controls';
+import SearchResultsSwitcher from './search-results-switcher';
+import MusicZone from './models/music-zone';
+import RadioPlayerConfig from './models/radio-player-config';
+
+@customElement('radio-player')
+export default class RadioPlayer extends LitElement {
+  @property({ type: RadioPlayerConfig }) config: RadioPlayerConfig | undefined = undefined;
+
+  @property({ type: TranscriptConfig }) transcriptConfig: TranscriptConfig | undefined = undefined;
+
+  @property({ type: Number }) currentTime = 0;
+
+  @property({ type: String }) searchTerm = '';
+
+  @property({ type: Boolean }) skipMusicSections = false;
+
+  @property({ type: Number }) private percentComplete = 0;
+
+  @property({ type: Boolean }) private isPlaying = false;
+
+  @property({ type: Number }) private duration = 0;
+
+  @property({ type: Number }) private playbackRate = 1;
+
+  @property({ type: Number }) private volume = 1;
+
+  @property({ type: Boolean }) private shouldShowSearchResultSwitcher = false;
+
+  @property({ type: Boolean }) private shouldShowNoSearchResultMessage = false;
+
+  private musicZones: MusicZone[] = [];
+
+  render(): TemplateResult {
+    return html`
+      ${this.audioElementTemplate}
+      <section role="main">
+        ${this.titleDateTemplate} ${this.collectionLogoTemplate} ${this.playbackControlsTemplate}
+        <div class="waveform-scrubber-container">
+          ${this.waveFormProgressTemplate} ${this.scrubberBarTemplate}
+        </div>
+        ${this.searchSectionTemplate} ${this.transcriptViewTemplate}
+      </section>
+    `;
+  }
+
+  play(): void {
+    /* istanbul ignore else */
```

Comment:
question: what is istanbul?

Reply:
Istanbul is the code coverage tool. It's pretty aggressive about its code coverage and if you have an `if` statement without a corresponding `else` clause, it complains that you haven't tested the `else` clause. 

In this case, we only want to call `play()` on the `audioElement` if it exists (it's nullable) but we don't care about the else clause.

This is another case that would get solved by TypeScript 3.7 (which just got released). I could have rewritten this line as: `this.audioElement?.play()`